src/rhsmlib/facts/hwprobe.py

```diff
@@ -782,20 +783,28 @@ def get_network_info(self) -> Dict[str, str]:
                 else:
                     net_info["network.fqdn"] = hostname
 
-            try:
-                info: List[tuple] = socket.getaddrinfo(hostname, None, socket.AF_INET, socket.SOCK_STREAM)
-                ip_list = set([x[4][0] for x in info])
-                net_info["network.ipv4_address"] = ", ".join(ip_list)
-            except Exception as err:
-                log.debug("Error during resolving IPv4 address of hostname: %s, %s" % (hostname, err))
+            ipv4_addresses: Set[str] = set()
+            ipv6_addresses: Set[str] = set()
+            for address_tuple in infolist:
+                address: str = address_tuple[4][0]
+                if address_tuple[0] is socket.AF_INET:
+                    ipv4_addresses.add(address)
+                if address_tuple[0] is socket.AF_INET6:
+                    ipv6_addresses.add(address)
+
+            # FIXME In 2017-09 (616ec72), we added `_get_ipvX_addr_list()` functions, but the original code
+            #  using sockets stayed; the new one was used just in the `except:` block. Why? Can't we just use
+            #  ethtool for everything? It is hard dependency and should always be available.
```

Comment:
sadly [python-ethtool](https://github.com/fedora-python/python-ethtool) seems to be deprecated too:
> This is the new upstream for python-ethtool maintained by Fedora's Python SIG. We ported it to Python 3 and only maintain it for the current tools to keep working. No new development is happening. This is a deprecated package. If you are considering to start using this, please don't. We recommend [netifaces](https://pypi.org/project/netifaces/) instead. Another alternative is [ethtool parsers](https://insights-core.readthedocs.io/en/latest/shared_parsers_catalog/ethtool.html) from [insights-core](https://pypi.org/project/insights-core/).

maybe it'd be better to use `ip` (via its JSON output) for the network-related facts, assuming it provides all the details we currently provide as facts